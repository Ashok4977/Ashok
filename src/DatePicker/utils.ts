import deepmerge from 'deepmerge'
import {DatePickerConfiguration, isMultiSelection, isRangeSelection, Selection} from './types'
import {defaultConfiguration} from './DatePickerProvider'

export const sanitizeDate = (date: Date | string) => new Date(new Date(date).toDateString())

export const getInitialFocusDate = (selection?: Selection) => {
  if (!selection) return sanitizeDate(new Date())

  if (selection instanceof Date) {
    return sanitizeDate(selection)
  } else if (isMultiSelection(selection)) {
    return sanitizeDate(selection[0])
  } else if (isRangeSelection(selection)) {
    return sanitizeDate(selection.from)
  } else {
    return sanitizeDate(new Date())
  }
}

export const initializeConfigurations = (externalConfig?: DatePickerConfiguration) => {
  let sanitizedConfig = {}
  if (externalConfig) {
    sanitizedConfig = Object.entries(externalConfig).filter(item => item[1] !== undefined && item[1] !== null)
    for (const item of sanitizedConfig) {
      if (item instanceof Date) {
        sanitizeDate(item)
      }
    }
  }
  return deepmerge(defaultConfiguration, sanitizedConfig)
}
