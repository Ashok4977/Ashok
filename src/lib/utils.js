
/**
 * Splice items from an Array for with the test function returns
 * truthy, modifying the original list and returning an array of
 * the spliced items.
 *
 * ```js
 * const children = React.Children.toArray(props.children)
 * const buttons = splice(children, child => child.type === Button)
 * ```
 */
const splice = (list, test) => {
  return list.reduce((acc, item, i) => {
    if (test(item)) {
      return acc.concat(list.splice(i, 1))
    }
    return acc
  }, [])
}

export {splice}
