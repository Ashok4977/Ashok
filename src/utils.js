import chroma from 'chroma-js'
import {css} from 'styled-components'
import {theme} from 'styled-system'

// color shortcuts
export const color = name => theme(`colors.${name}`)
export const black = color('black')
export const white = color('white')

/**
 * Reduce a value to a primitive as long as it is a function.
 *
 * interpolate(() => 'x') // returns 'x'
 * interpolate(() => () => 'y') // returns 'y'
 */
export const interpolate = (value, context, props) => {
  while (typeof value === 'function') {
    value = value.call(context, props)
  }
  return value
}

/**
 * This decorator interpolates all of the function's arguments by calling any
 * functions in the provided context and with the provided arguments. In other
 * words, the function you provide is guaranteed to have all of its arguments
 * "called" if they are functions. Example:
 *
 * const rgba = interpolator((rgb, a) => chroma(rgb).alpha(a).css())
 * const fadeed = rgba(theme('colors.black'), .5) // returns a function!
 * faded({}) // probably throws an error because props.theme isn't set
 */
export const interpolator = f => {
  return (...outer) => {
    return function(props) {
      return f(...[...outer].map(a => interpolate(a, this, props)))
    }
  }
}

export const rgba = interpolator(       (rgb, alpha)    => chroma(rgb).alpha(alpha).css())
export const lighten = interpolator(    (value, amount) => chroma(value).brighten(amount).css())
export const darken = interpolator(     (value, amount) => chroma(value).darken(amount).css())
export const desaturate = interpolator( (value, amount) => chroma(value).desaturate(amount).css())
export const mix = interpolator(        (a, b, amount)  => chroma.mix(a, b, amount).css())
