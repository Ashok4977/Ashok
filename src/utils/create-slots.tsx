import React from 'react'
import useLayoutEffect from './useIsomorphicLayoutEffect'

/** createSlots is a factory that can create a
 *  typesafe Slots + Slot pair to use in a component definition
 *  For example: ActionList.Item uses createSlots to get a Slots wrapper
 *  + Slot component that is used by LeadingVisual, Description
 */
const createSlots = <SlotNames extends string>(_slotNames: SlotNames[]) => {
  type Slots = {
    [key in SlotNames]?: React.ReactNode
  }

  type ContextProps = {
    registerSlot: (name: SlotNames, contents: React.ReactNode) => void
    unregisterSlot: (name: SlotNames) => void
  }
  const SlotsContext = React.createContext<ContextProps>({
    registerSlot: () => null,
    unregisterSlot: () => null
  })

  const Slots: React.FC<{
    children: (slots: Slots) => React.ReactNode
  }> = ({children}) => {
    const [slots, setSlots] = React.useState<Slots>({})

    const registerSlot = React.useCallback(
      (name: SlotNames, contents: React.ReactNode) => setSlots(prevSlots => ({...prevSlots, [name]: contents})),
      []
    )

    // Slot can be removed from the tree as well,
    // we need to unregister them from the slot
    const unregisterSlot = React.useCallback(
      (name: SlotNames) =>
        setSlots(prevSlots => {
          const newSlots = {...prevSlots}
          delete newSlots[name]
          return newSlots
        }),
      []
    )

    return <SlotsContext.Provider value={{registerSlot, unregisterSlot}}>{children(slots)}</SlotsContext.Provider>
  }

  const Slot: React.FC<{
    name: SlotNames
    children: React.ReactNode
  }> = ({name, children}) => {
    const {registerSlot, unregisterSlot} = React.useContext(SlotsContext)

    useLayoutEffect(() => {
      registerSlot(name, children)
      return () => unregisterSlot(name)
    }, [name, children, registerSlot, unregisterSlot])

    return null
  }

  return {Slots, Slot}
}

export default createSlots
