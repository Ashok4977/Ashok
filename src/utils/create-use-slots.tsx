import React from 'react'
import {useForceUpdate} from './use-force-update'

/** Create useSlots is a factory that can create a typed
 *  useSlots + Slot component combination for a component
 *  For example: ActionList uses createUseSlots to get a
 *  useSlot hook that can be used inside each Item
 */
export const createUseSlots = <SlotName extends string>(slotNames: SlotName[]) => {
  type ContextType = {
    registerSlot: (name: SlotName, contents: React.ReactNode) => void
    unregisterSlot: (name: SlotName) => void
  }
  const SlotsContext = React.createContext<ContextType>({registerSlot: () => null, unregisterSlot: () => null})

  const useSlots = () => {
    // Double render strategy
    // when the effect is run for the first time,
    // all the children have mounted = registed themself in slot.
    // we re-render the Item component to re-render with filled slots.
    const rerenderWithSlots = useForceUpdate()
    const [isMounted, setIsMounted] = React.useState(false)

    const slotsDefinition: {[key in SlotName]?: JSX.Element | null} = {}
    slotNames.map(name => {
      slotsDefinition[name] = null
    })
    const slotsRef = React.useRef<{[key in SlotName]?: React.ReactNode}>(slotsDefinition)

    // fires once after all the slots are registered
    React.useLayoutEffect(() => {
      setIsMounted(true)
      rerenderWithSlots()
    }, [rerenderWithSlots])

    const registerSlot = React.useCallback(
      (name: SlotName, contents: React.ReactNode) => {
        slotsRef.current[name] = contents

        // if something has changed?

        // don't render until the component mounts = all slots are registered
        if (isMounted) rerenderWithSlots()
      },
      [isMounted, rerenderWithSlots]
    )

    // Item.* can be removed from the DOM,
    // we need to unregister them from the slot
    const unregisterSlot = React.useCallback(
      (name: SlotName) => {
        slotsRef.current[name] = null
        rerenderWithSlots()
      },
      [rerenderWithSlots]
    )

    const SlotsProvider: React.FC = ({children}) => {
      return <SlotsContext.Provider value={{registerSlot, unregisterSlot}}>{children}</SlotsContext.Provider>
    }

    return {slots: slotsRef.current, SlotsProvider}
  }

  const Slot: React.FC<{name: SlotName}> = ({name, children}) => {
    const {registerSlot, unregisterSlot} = React.useContext(SlotsContext)

    React.useLayoutEffect(() => {
      registerSlot(name, children)
      return () => unregisterSlot(name)
      // registerSlot and unregisterSlot are created by the ItemContext,
      // we can safely ignore them because they will not change between renders
      // eslint-disable-next-line react-hooks/exhaustive-deps
    }, [name, children])

    return null
  }

  return {useSlots, Slot}
}
