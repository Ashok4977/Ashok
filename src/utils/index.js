
export function defined(val) {
  return val !== null && typeof val !== 'undefined'
}

export function definedFallback(input, fallback) {
  return defined(input) ? input : fallback
}
