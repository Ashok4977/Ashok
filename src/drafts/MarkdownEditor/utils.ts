import {isMacOS} from '@primer/behaviors/utils'

export type InputState = {
  readonly value: string
  readonly selectionStart: number
  readonly selectionEnd: number
}

export const getSelectedLineRange = (inputState: InputState): [number, number] => {
  // Subtract one from the caret position so the newline found is not the one _at_ the caret position
  // then add one because we don't want to include the found newline. Also changes -1 (not found) result to 0
  const start = inputState.value.lastIndexOf('\n', inputState.selectionStart - 1) + 1

  // activeLineEnd will be the index of the next newline inclusive, which works because slice is last-index exclusive
  let end = inputState.value.indexOf('\n', inputState.selectionEnd)
  if (end === -1) end = inputState.value.length

  return [start, end]
}

export const markdownComment = (text: string) => `<!-- ${text.replaceAll('--', '\\-\\-')} -->`

export const markdownLink = (text: string, url: string) =>
  `[${text.replaceAll('[', '\\[').replaceAll(']', '\\]')}](${url.replaceAll('(', '\\(').replaceAll(')', '\\)')})`

export const markdownImage = (altText: string, url: string) => `!${markdownLink(altText, url)}`

export const isModifierKey = (event: KeyboardEvent | React.KeyboardEvent<unknown>) =>
  isMacOS() ? event.metaKey : event.ctrlKey

export function isMultipleLines(string: string): boolean {
  return string.trim().split('\n').length > 1
}
