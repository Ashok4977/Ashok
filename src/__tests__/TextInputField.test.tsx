import React from 'react'
import {render} from '../utils/testing'
import {render as HTMLRender, cleanup} from '@testing-library/react'
import {axe, toHaveNoViolations} from 'jest-axe'
import 'babel-polyfill'
import {TextInputField, TextInputWithTokens} from '..'
expect.extend(toHaveNoViolations)

const TEXTINPUTFIELD_LABEL_TEXT = 'Name'
const TEXTINPUTFIELD_CAPTION_TEXT = 'Hint: your first name'
const TEXTINPUTFIELD_SUCCESS_TEXT = 'This name is valid'

describe('TextInputField', () => {
  describe('snapshots', () => {
    it('renders with a label an input', () => {
      expect(
        render(
          <TextInputField>
            <TextInputField.Label>{TEXTINPUTFIELD_LABEL_TEXT}</TextInputField.Label>
            <TextInputField.Input />
          </TextInputField>
        )
      ).toMatchSnapshot()
    })
    it('renders with a hidden label', () => {
      expect(
        render(
          <TextInputField>
            <TextInputField.Label visuallyHidden>{TEXTINPUTFIELD_LABEL_TEXT}</TextInputField.Label>
            <TextInputField.Input />
          </TextInputField>
        )
      ).toMatchSnapshot()
    })
    it('renders with a custom ID', () => {
      expect(
        render(
          <TextInputField id="customId">
            <TextInputField.Label>{TEXTINPUTFIELD_LABEL_TEXT}</TextInputField.Label>
            <TextInputField.Input />
          </TextInputField>
        )
      ).toMatchSnapshot()
    })
    it('renders as disabled', () => {
      expect(
        render(
          <TextInputField disabled>
            <TextInputField.Label visuallyHidden>{TEXTINPUTFIELD_LABEL_TEXT}</TextInputField.Label>
            <TextInputField.Input />
          </TextInputField>
        )
      ).toMatchSnapshot()
    })
    it('renders as required', () => {
      expect(
        render(
          <TextInputField required>
            <TextInputField.Label>{TEXTINPUTFIELD_LABEL_TEXT}</TextInputField.Label>
            <TextInputField.Input />
          </TextInputField>
        )
      ).toMatchSnapshot()
    })
    it('renders with a caption', () => {
      expect(
        render(
          <TextInputField>
            <TextInputField.Label>{TEXTINPUTFIELD_LABEL_TEXT}</TextInputField.Label>
            <TextInputField.Input />
            <TextInputField.Caption>{TEXTINPUTFIELD_CAPTION_TEXT}</TextInputField.Caption>
          </TextInputField>
        )
      ).toMatchSnapshot()
    })
    it('renders with a successful validation message', () => {
      expect(
        render(
          <TextInputField
            validationMap={{
              noSpaces: 'error',
              validName: 'success'
            }}
            validationResult="validName"
          >
            <TextInputField.Label>{TEXTINPUTFIELD_LABEL_TEXT}</TextInputField.Label>
            <TextInputField.Input />
            <TextInputField.Validation validationKey="validName">
              {TEXTINPUTFIELD_SUCCESS_TEXT}
            </TextInputField.Validation>
          </TextInputField>
        )
      ).toMatchSnapshot()
    })
    it('renders with an error validation message', () => {
      expect(
        render(
          <TextInputField
            validationMap={{
              noSpaces: 'error',
              validName: 'success'
            }}
            validationResult="noSpaces"
          >
            <TextInputField.Label>{TEXTINPUTFIELD_LABEL_TEXT}</TextInputField.Label>
            <TextInputField.Input />
            <TextInputField.Validation validationKey="noSpaces">
              Your first name cannot contain spaces
            </TextInputField.Validation>
          </TextInputField>
        )
      ).toMatchSnapshot()
    })
    it('renders with the input as a TextInputWithTokens', () => {
      const onRemoveMock = jest.fn()
      expect(
        render(
          <TextInputField>
            <TextInputField.Label>{TEXTINPUTFIELD_LABEL_TEXT}</TextInputField.Label>
            <TextInputField.Input
              as={TextInputWithTokens}
              tokens={[
                {text: 'zero', id: 0},
                {text: 'one', id: 1},
                {text: 'two', id: 2}
              ]}
              onRemove={onRemoveMock}
            />
          </TextInputField>
        )
      ).toMatchSnapshot()
    })
  })

  describe('ARIA attributes', () => {
    it('associates the label with the input', () => {
      const {getByLabelText} = HTMLRender(
        <TextInputField>
          <TextInputField.Label>{TEXTINPUTFIELD_LABEL_TEXT}</TextInputField.Label>
          <TextInputField.Input />
        </TextInputField>
      )

      const inputNode = getByLabelText(TEXTINPUTFIELD_LABEL_TEXT)
      expect(inputNode).toBeDefined()
    })
    it('associates caption text with the input', () => {
      const fieldId = 'captionedInput'
      const {getByLabelText, getByText} = HTMLRender(
        <TextInputField id={fieldId}>
          <TextInputField.Label>{TEXTINPUTFIELD_LABEL_TEXT}</TextInputField.Label>
          <TextInputField.Input />
          <TextInputField.Caption>{TEXTINPUTFIELD_CAPTION_TEXT}</TextInputField.Caption>
        </TextInputField>
      )

      const inputNode = getByLabelText(TEXTINPUTFIELD_LABEL_TEXT)
      const captionNode = getByText(TEXTINPUTFIELD_CAPTION_TEXT)

      expect(captionNode.getAttribute('id')).toBe(`${fieldId}-caption`)
      expect(inputNode.getAttribute('aria-describedby')).toBe(`${fieldId}-caption`)
    })
    it('associates validation text with the input', () => {
      const fieldId = 'validatedInput'
      const {getByLabelText, getByText} = HTMLRender(
        <TextInputField id={fieldId} validationMap={{validName: 'success'}} validationResult="validName">
          <TextInputField.Label>{TEXTINPUTFIELD_LABEL_TEXT}</TextInputField.Label>
          <TextInputField.Input />
          <TextInputField.Validation validationKey="validName">{TEXTINPUTFIELD_SUCCESS_TEXT}</TextInputField.Validation>
        </TextInputField>
      )

      const inputNode = getByLabelText(TEXTINPUTFIELD_LABEL_TEXT)
      const validationNode = getByText(TEXTINPUTFIELD_SUCCESS_TEXT)

      expect(validationNode.getAttribute('id')).toBe(`${fieldId}-validationMsg`)
      expect(inputNode.getAttribute('aria-describedby')).toBe(`${fieldId}-validationMsg`)
    })
  })

  it('should have no axe violations', async () => {
    const {container} = HTMLRender(
      <TextInputField>
        <TextInputField.Label>{TEXTINPUTFIELD_LABEL_TEXT}</TextInputField.Label>
        <TextInputField.Input />
        <TextInputField.Caption>{TEXTINPUTFIELD_CAPTION_TEXT}</TextInputField.Caption>
      </TextInputField>
    )
    const results = await axe(container)
    expect(results).toHaveNoViolations()
    cleanup()
  })
})
