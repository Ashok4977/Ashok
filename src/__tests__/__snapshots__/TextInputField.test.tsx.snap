// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInputField snapshots renders as disabled 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c0 > *:not(label) + * {
  margin-top: 8px;
}

.c3 {
  border: 0;
  font-size: inherit;
  font-family: inherit;
  background-color: transparent;
  -webkit-appearance: none;
  color: inherit;
  width: 100%;
}

.c3:focus {
  outline: 0;
}

.c2 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  min-height: 34px;
  font-size: 14px;
  line-height: 20px;
  color: #24292f;
  vertical-align: middle;
  background-repeat: no-repeat;
  background-position: right 8px center;
  border: 1px solid #d0d7de;
  border-radius: 6px;
  outline: none;
  box-shadow: inset 0 1px 0 rgba(208,215,222,0.2);
  cursor: text;
  padding: 6px 12px;
  color: #8c959f;
  background-color: rgba(175,184,193,0.2);
  border-color: #d0d7de;
}

.c2 .TextInput-icon {
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  color: #57606a;
  margin: 0 8px;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.c2:focus-within {
  border-color: #0969da;
  box-shadow: 0 0 0 3px rgba(9,105,218,0.3);
}

.c1 {
  position: absolute;
  width: 1px;
  height: 1px;
  padding: 0;
  margin: -1px;
  overflow: hidden;
  -webkit-clip: rect(0,0,0,0);
  clip: rect(0,0,0,0);
  white-space: nowrap;
  border-width: 0;
}

@media (min-width:768px) {
  .c2 {
    font-size: 14px;
  }
}

<div
  className="c0"
  display="flex"
  width="100%"
>
  <label
    className="c1"
    htmlFor="__primer_id_10002"
  >
    Name
  </label>
  <span
    className="c2 TextInput-wrapper"
    disabled={true}
  >
    <input
      aria-describedby=""
      className="c3"
      disabled={true}
      id="__primer_id_10002"
      type="text"
    />
  </span>
</div>
`;

exports[`TextInputField snapshots renders as required 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c0 > * + * {
  margin-top: 8px;
}

<div
  className="c0"
  display="flex"
  width="100%"
/>
`;

exports[`TextInputField snapshots renders with a caption 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c0 > * + * {
  margin-top: 8px;
}

.c3 {
  border: 0;
  font-size: inherit;
  font-family: inherit;
  background-color: transparent;
  -webkit-appearance: none;
  color: inherit;
  width: 100%;
}

.c3:focus {
  outline: 0;
}

.c2 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  min-height: 34px;
  font-size: 14px;
  line-height: 20px;
  color: #24292f;
  vertical-align: middle;
  background-repeat: no-repeat;
  background-position: right 8px center;
  border: 1px solid #d0d7de;
  border-radius: 6px;
  outline: none;
  box-shadow: inset 0 1px 0 rgba(208,215,222,0.2);
  cursor: text;
  padding: 6px 12px;
}

.c2 .TextInput-icon {
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  color: #57606a;
  margin: 0 8px;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.c2:focus-within {
  border-color: #0969da;
  box-shadow: 0 0 0 3px rgba(9,105,218,0.3);
}

.c1 {
  font-weight: 600;
  font-size: 14px;
  display: block;
  color: #24292f;
}

.c4 {
  font-size: 12px;
  color: #57606a;
  display: block;
}

@media (min-width:768px) {
  .c2 {
    font-size: 14px;
  }
}

<div
  className="c0"
  display="flex"
  width="100%"
>
  <label
    className="c1"
    htmlFor="__primer_id_10004"
  >
    Name
  </label>
  <span
    className="c2 TextInput-wrapper"
  >
    <input
      aria-describedby="__primer_id_10004-caption"
      className="c3"
      id="__primer_id_10004"
      type="text"
    />
  </span>
  <span
    className="c4"
    color="fg.muted"
    display="block"
    fontSize={0}
    id="__primer_id_10004-caption"
  >
    Hint: your first name
  </span>
</div>
`;

exports[`TextInputField snapshots renders with a custom ID 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c0 > * + * {
  margin-top: 8px;
}

<div
  className="c0"
  display="flex"
  width="100%"
/>
`;

exports[`TextInputField snapshots renders with a hidden label 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c0 > *:not(label) + * {
  margin-top: 8px;
}

.c3 {
  border: 0;
  font-size: inherit;
  font-family: inherit;
  background-color: transparent;
  -webkit-appearance: none;
  color: inherit;
  width: 100%;
}

.c3:focus {
  outline: 0;
}

.c2 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  min-height: 34px;
  font-size: 14px;
  line-height: 20px;
  color: #24292f;
  vertical-align: middle;
  background-repeat: no-repeat;
  background-position: right 8px center;
  border: 1px solid #d0d7de;
  border-radius: 6px;
  outline: none;
  box-shadow: inset 0 1px 0 rgba(208,215,222,0.2);
  cursor: text;
  padding: 6px 12px;
}

.c2 .TextInput-icon {
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  color: #57606a;
  margin: 0 8px;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.c2:focus-within {
  border-color: #0969da;
  box-shadow: 0 0 0 3px rgba(9,105,218,0.3);
}

.c1 {
  position: absolute;
  width: 1px;
  height: 1px;
  padding: 0;
  margin: -1px;
  overflow: hidden;
  -webkit-clip: rect(0,0,0,0);
  clip: rect(0,0,0,0);
  white-space: nowrap;
  border-width: 0;
}

@media (min-width:768px) {
  .c2 {
    font-size: 14px;
  }
}

<div
  className="c0"
  display="flex"
  width="100%"
>
  <label
    className="c1"
    htmlFor="__primer_id_10001"
  >
    Name
  </label>
  <span
    className="c2 TextInput-wrapper"
  >
    <input
      aria-describedby=""
      className="c3"
      id="__primer_id_10001"
      type="text"
    />
  </span>
</div>
`;

exports[`TextInputField snapshots renders with a label an input 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c0 > * + * {
  margin-top: 8px;
}

<div
  className="c0"
  display="flex"
  width="100%"
/>
`;

exports[`TextInputField snapshots renders with a successful validation message 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c0 > * + * {
  margin-top: 8px;
}

.c4 {
  margin-right: 4px;
}

.c1 {
  height: auto;
  overflow: hidden;
}

.c2 {
  -webkit-animation: 170ms eGcHP cubic-bezier(0.44,0.74,0.36,1);
  animation: 170ms eGcHP cubic-bezier(0.44,0.74,0.36,1);
}

.c3 {
  font-size: 12px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  color: #1a7f37;
  -webkit-align-items: baseline;
  -webkit-box-align: baseline;
  -ms-flex-align: baseline;
  align-items: baseline;
}

.c3 a {
  color: currentColor;
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

<div
  className="c0"
  display="flex"
  width="100%"
>
  <div
    className="c1"
    height="auto"
    overflow="hidden"
  >
    <div
      className="c2"
      onAnimationEnd={[Function]}
    >
      <span
        className="c3"
        color="success.fg"
        display="flex"
        fontSize={0}
      >
        <span
          className="c4"
        >
          <svg
            aria-hidden="true"
            className="octicon octicon-check-circle-fill"
            dangerouslySetInnerHTML={
              Object {
                "__html": "<path fill-rule=\\"evenodd\\" d=\\"M6 0a6 6 0 100 12A6 6 0 006 0zm-.705 8.737L9.63 4.403 8.392 3.166 5.295 6.263l-1.7-1.702L2.356 5.8l2.938 2.938z\\"></path>",
              }
            }
            fill="currentColor"
            height={12}
            role="img"
            style={
              Object {
                "display": "inline-block",
                "overflow": "visible",
                "userSelect": "none",
                "verticalAlign": "text-bottom",
              }
            }
            viewBox="0 0 12 12"
            width={12}
          />
        </span>
        <span
          id="__primer_id_10005-validationMsg"
        >
          This name is valid
        </span>
      </span>
    </div>
  </div>
</div>
`;

exports[`TextInputField snapshots renders with an error validation message 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c0 > * + * {
  margin-top: 8px;
}

.c7 {
  margin-right: 4px;
}

.c4 {
  height: auto;
  overflow: hidden;
}

.c3 {
  border: 0;
  font-size: inherit;
  font-family: inherit;
  background-color: transparent;
  -webkit-appearance: none;
  color: inherit;
  width: 100%;
}

.c3:focus {
  outline: 0;
}

.c2 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  min-height: 34px;
  font-size: 14px;
  line-height: 20px;
  color: #24292f;
  vertical-align: middle;
  background-repeat: no-repeat;
  background-position: right 8px center;
  border: 1px solid #d0d7de;
  border-radius: 6px;
  outline: none;
  box-shadow: inset 0 1px 0 rgba(208,215,222,0.2);
  cursor: text;
  padding: 6px 12px;
}

.c2 .TextInput-icon {
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  color: #57606a;
  margin: 0 8px;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.c2:focus-within {
  border-color: #0969da;
  box-shadow: 0 0 0 3px rgba(9,105,218,0.3);
}

.c1 {
  font-weight: 600;
  font-size: 14px;
  display: block;
  color: #24292f;
}

.c5 {
  -webkit-animation: 170ms eGcHP cubic-bezier(0.44,0.74,0.36,1);
  animation: 170ms eGcHP cubic-bezier(0.44,0.74,0.36,1);
}

.c6 {
  font-size: 12px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  color: #cf222e;
  -webkit-align-items: baseline;
  -webkit-box-align: baseline;
  -ms-flex-align: baseline;
  align-items: baseline;
}

.c6 a {
  color: currentColor;
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

@media (min-width:768px) {
  .c2 {
    font-size: 14px;
  }
}

<div
  className="c0"
  display="flex"
  width="100%"
>
  <label
    className="c1"
    htmlFor="__primer_id_10006"
  >
    Name
  </label>
  <span
    className="c2 TextInput-wrapper"
  >
    <input
      aria-describedby="__primer_id_10006-validationMsg"
      className="c3"
      id="__primer_id_10006"
      type="text"
    />
  </span>
  <div
    className="c4"
    height="auto"
    overflow="hidden"
  >
    <div
      className="c5"
      onAnimationEnd={[Function]}
    >
      <span
        className="c6"
        color="danger.fg"
        display="flex"
        fontSize={0}
      >
        <span
          className="c7"
        >
          <svg
            aria-hidden="true"
            className="octicon octicon-alert-fill"
            dangerouslySetInnerHTML={
              Object {
                "__html": "<path fill-rule=\\"evenodd\\" d=\\"M4.855.708c.5-.896 1.79-.896 2.29 0l4.675 8.351a1.312 1.312 0 01-1.146 1.954H1.33A1.312 1.312 0 01.183 9.058L4.855.708zM7 7V3H5v4h2zm-1 3a1 1 0 100-2 1 1 0 000 2z\\"></path>",
              }
            }
            fill="currentColor"
            height={12}
            role="img"
            style={
              Object {
                "display": "inline-block",
                "overflow": "visible",
                "userSelect": "none",
                "verticalAlign": "text-bottom",
              }
            }
            viewBox="0 0 12 12"
            width={12}
          />
        </span>
        <span
          id="__primer_id_10006-validationMsg"
        >
          Your first name cannot contain spaces
        </span>
      </span>
    </div>
  </div>
</div>
`;

exports[`TextInputField snapshots renders with the input as a TextInputWithTokens 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c0 > * + * {
  margin-top: 8px;
}

<div
  className="c0"
  display="flex"
  width="100%"
/>
`;
